src/libertem/udf/raw.py

```diff
@@ -75,3 +76,107 @@ def merge_all(self, ordered_results):
         intensity_chunks = [b.intensity for b in ordered_results.values()]
         intensity_sum = np.stack(intensity_chunks, axis=0).sum(axis=0)
         return {'intensity': intensity_sum}
+
+
+class PickShiftedUDF(UDF):
+    '''
+    Load raw data from ROI and applies a corrective shift.
+
+    This UDF is useful to compensate for descan, separating the
+    effect from material characteristics.
+
+    Parameters
+    ----------
+    regression_coefficients : (3, 2) matrix
+        A vector of regression coefficients for x- and y-directions.
+        The regression coefficients make up two planes that describe
+        the descan and are used to remove it.
+        The coefficients are arranged in the following manner:
+        ((y, x), (dy/dy, dx/dy), (dy/dx, dx/dx)) where y and x are
+        the shifts at coordinates (0, 0). The displayed image will be
+        shifted in x- and y-directions by these parameters. By default
+        `None`, which corresponds to no shifts.
+    '''
```

Comment:
What about supporting the same options as `CoMUDF`, or even use the `CoMParams` class: https://github.com/LiberTEM/LiberTEM/blob/59529c19ca4fcaac63aef25d524c80e3389e97c9/src/libertem/udf/com.py#L30-L44

`scan_rotation` and `flip_y` would apply perfectly for this use case and should be easy to implement with the `scipy.ndimage` functions.

In particular when dealing with DPC, it can help if in the picked frames everything points in the right directions relative to the scan! By using the same parameters, one can use them for both. As a side effect one could use this to create a corrected dataset! Could be merged with corresponding code in pyxem as a perspective?